src/workerd/io/trace-stream.c++

```diff
@@ -144,7 +144,13 @@ jsg::JsValue ToJs(jsg::Lock& js, const tracing::Attribute::Value& value) {
       return js.num(d);
     }
     KJ_CASE_ONEOF(i, int64_t) {
-      return js.bigInt(i);
+      // We report integer values using the JS number type, instead of BigInt (which is not natively
+      // stringifiable). This limits precision to 53 bits.
+      // TODO: Document this limitation
+      // TODO(someday): Prohibit reporting integer tag values at tag creation time? For that we'd
+      // need to change the SpanTag definition, not sure if it's it since being unable to create a
+      // tag with an integral type is confusing.
+      return js.num((double)i);
```

Comment:
Note that this conversion is lossy if the value of `i` is large enough to cause the value to start losing precision. Perhaps we should consider stringifying instead?

Reply:
Perhaps we can emit a warning to sentry if converting `i` would lose precision so that we at least get a heads up on the off chance it ever does happen. We can re-evaluate the conversion if we ever start seeing those logs.

Reply:
@danlapid discussed the PR with @mar-cf – he thinks additional discussion is needed here, beyond what we discussed in chat. Do you think this is the right approach?
An example for why I think number is right here: If we want to signify what the size in bytes of a return response is, we'd want that represented as a number (without needing to convert it from a string). In this case (and I believe the vast majority of use cases) we do not expect that length to be larger than 2^53. As noted, we still need to document that limitation. For cases where we expect numbers larger than 2^53 to be realistic (should match none of the tags we currently have), we'll provide a string value at tag creation time instead.